internal/adapter/lsp/document.go

```diff
@@ -259,10 +313,17 @@ func (d *document) DocumentLinks() ([]documentLink, error) {
 		}
 
 		for _, match := range wikiLinkRegex.FindAllStringSubmatchIndex(line, -1) {
+			// Ignore when inside backticks: `[[filename]]`
+			if linkWithinInlineCode(line, match[0], match[1], insideInline) {
+				continue
+			}
 			href := line[match[2]:match[3]]
 			hasTitle := match[4] != -1
 			appendLink(href, match[0], match[1], hasTitle, true)
 		}
+		if strings.Count(line, "`") % 2 == 1 {
+			insideInline = !insideInline
+		}
```

Comment:
This is taking into account inline code over a line break right?
Is there a reason to assign the inverse here, instead of just `true`?

As I'm thinking if there are an odd number of backticks, the next line should _always_ begin with `insideInline = true` or?

I'm also wondering if there's ever a case where the line starts with `insideInline = true`, and there being an odd number of backticks. This would then make `insideInline = false` for the subsequent line.

Reply:
addressed [in comment](https://github.com/zk-org/zk/pull/399#issuecomment-2039003713)